backend/de.metas.payment.sepa/base/src/main/java/de/metas/payment/sepa/model/validator/C_BP_BankAccount.java

```diff
@@ -74,4 +78,35 @@ public void validateIBAN(final I_C_BP_BankAccount bp_bankAccount)
 
 	}
 
+	@ModelChange(
+			timings = { ModelValidator.TYPE_BEFORE_CHANGE, ModelValidator.TYPE_AFTER_NEW },
+			ifColumnsChanged = { I_C_BP_BankAccount.COLUMNNAME_A_Country_ID}
+	)
+	public void setA_Country(final I_C_BP_BankAccount bp_bankAccount)
+	{
+		final int countryId = bp_bankAccount.getA_Country_ID();
+		if (countryId > 0)
+		{
+			final String countryCode = countryDAO.getCountryCode(CountryId.ofRepoId(countryId));
+			bp_bankAccount.setA_Country(countryCode);
+		}
+	}
+
+	@ModelChange(
+			timings = { ModelValidator.TYPE_BEFORE_CHANGE, ModelValidator.TYPE_AFTER_NEW },
+			ifColumnsChanged = { I_C_BP_BankAccount.COLUMNNAME_A_Country}
+	)
+	public void validateA_Country(final I_C_BP_BankAccount bp_bankAccount)
+	{
+		final String A_Country = bp_bankAccount.getA_Country();
+		if (StringUtils.trimBlankToNull(A_Country) != null)
+		{
+			final CountryId countryId = countryDAO.getCountryIdByCountryCodeOrNull(A_Country);
+			if (countryId == null)
```

Comment:
how is the A_Country_ID set if if was found using the code?

Reply:
We do not set it back if it was via API, by setting A_Country.
We validate that A_Country is a country code, regardless of how it was set.